src/main/themes.ts

```diff
@@ -0,0 +1,165 @@
+import { render } from 'sass'
+import { readFile, existsSync, watchFile, readdir } from 'fs-extra'
+import { join, basename } from 'path'
+import { Theme } from '../shared/shared-types'
+import { getCustomThemesPath } from './application-constants'
+import { app as rawApp, systemPreferences } from 'electron'
+import { ExtendedAppMainProcess } from './types'
+import * as mainWindow from './windows/main'
+
+const app = rawApp as ExtendedAppMainProcess
+
+import { getLogger } from '../shared/logger'
+const log = getLogger('main/themes')
+
+const dc_theme_dir = join(__dirname, '../../themes')
+
+function parseThemeMetaData(
+  rawTheme: string
+): { name: string; description: string } {
+  const meta_data_block = /.theme-meta {([^]*)}/gm.exec(rawTheme)[1].trim()
+
+  const regex = /--(\w*): ?['"]([^]*?)['"];?/gi
+
+  let meta: { [key: string]: string } = {}
+
+  let last_result: any = true
+
+  while (last_result) {
+    last_result = regex.exec(meta_data_block)
+    if (last_result) {
+      meta[last_result[1]] = last_result[2]
+    }
+  }
+
+  // check if name and description are defined
+  if (!meta.name || !meta.description) {
+    throw new Error(
+      'The meta variables meta.name and meta.description must be defined'
+    )
+  }
+
+  return <any>meta
+}
+
+async function readThemeDir(path: string, prefix: string): Promise<Theme[]> {
+  let files = await readdir(path)
+  return Promise.all(
+    files
+      .filter(f => f.includes('.scss') && f.charAt(0) !== '_')
+      .map(async f => {
+        const theme_meta = parseThemeMetaData(
+          await readFile(join(path, f), 'utf-8')
+        )
+        return {
+          name: theme_meta.name,
+          description: theme_meta.description,
+          address: prefix + ':' + basename(f, '.scss'),
+        }
+      })
+  )
+}
+
+export async function getAvailibleThemes(): Promise<Theme[]> {
+  // look at the common places for themes
+  return [
+    ...(await readThemeDir(dc_theme_dir, 'dc')),
+    ...(await readThemeDir(getCustomThemesPath(), 'custom')),
+  ]
+}
+
+export async function loadTheme(
+  theme_path: string
```

Comment:
I'm against having a theme_path. I'm fine with having a theme name and two locations from where we load them. Either the installed ones, or from "~/.config/DeltaChat/themes". Everything else is a possible LFI vulnerability.